src/contexts/addresses.tsx

```diff
@@ -233,44 +228,57 @@ export const AddressesContextProvider: FC<{ overrideContextValue?: PartialDeep<A
     [wallet, activeWalletName, isPassphraseUsed, setAddress]
   )
 
+  const displayDataFetchingError = useCallback(
+    () =>
+      setSnackbarMessage({
+        text: t`Could not fetch data because the wallet is offline`,
+        type: 'alert',
+        duration: 5000
+      }),
+    [setSnackbarMessage, t]
+  )
+
   const fetchPendingTxs = useCallback(
     async (addresses: Address[] = []) => {
       if (!client) {
-        setSnackbarMessage({
-          text: t`Could not fetch data because the wallet is offline`,
-          type: 'alert',
-          duration: 5000
-        })
+        displayDataFetchingError()
         return
       }
+      setIsLoadingData(true)
 
-      // Make a sequence of requests from the addresses to not spam the explorer API
-      return addresses.reduce(
-        (p, address) =>
-          p.then(() =>
-            client.explorer.addresses.getAddressesAddressUnconfirmedTransactions(address.hash).then((response) => {
-              const { data: txs } = response
-              txs.forEach((tx) => {
-                if (tx.type !== 'Unconfirmed') return
-                if (address.transactions.pending.some((t: PendingTx) => t.txId == tx.hash)) return
-                address.addPendingTransaction(fromUnconfirmedTransactionToPendingTx(tx, address.hash, currentNetwork))
-              })
-            })
-          ),
-        Promise.resolve()
-      )
```

Comment:
Looking at the devtools, the requests were not made in a sequence (one after the other resolves), but all together. The updated logic is similar to how `fetchAndStoreAddressesData` is implemented.

Reply:
How is that, when it's a chain of `.then()` ? (Haven't verified but it seemed sequential to me)

Reply:
It didn't look sequential to me when I tested.

In any case, I think that the change I made makes the code similar to existing code (using await syntax)

Reply:
Doesn't matter much to me which way we encode the "sequencing" of the promises - I do it the other way because it's a common idiom (plus I find the Promise API superior in pretty much every way to async/await, but that's another conversation to have elsewhere :p)

As long as the new code is equiv. all good :+1: